rcnn/BatchedNms.cu

```diff
@@ -0,0 +1,125 @@
+#include "BatchedNmsPlugin.h"
+#include "cuda_utils.h"
+
+#include <algorithm>
+#include <iostream>
+#include <stdexcept>
+#include <cstdint>
+#include <vector>
+#include <cmath>
+
+#include <cuda.h>
+#include <thrust/device_ptr.h>
+#include <thrust/sequence.h>
+#include <thrust/execution_policy.h>
+#include <thrust/gather.h>
+#include <thrust/system/cuda/detail/cub/device/device_radix_sort.cuh>
+
+namespace nvinfer1 {
+
+__global__ void batched_nms_kernel(
+	const int num_per_thread, const float threshold, const int num_detections,
+	const int *indices, float *scores, const float *classes, const float4 *boxes) {
+
+	// Go through detections by descending score
+	for (int m = 0; m < num_detections; m++) {
+		for (int n = 0; n < num_per_thread; n++) {
+			int i = threadIdx.x * num_per_thread + n;
+			if (i < num_detections && m < i && scores[m] > 0.0f) {
+				int idx = indices[i];
+				int max_idx = indices[m];
+				int icls = classes[idx];
```

Comment:
can you 4-space instead of tab? for all.

Reply:
Thanks, i'll adjust that.